### PR TITLE
backend: Make email as pk for CustomUser

### PR DESCRIPTION
**Pull request by iamkroot**

### CODE REVIEW

File: backend/users/auth.py [hunk 1]
```diff
@@ -13,9 +13,6 @@ class CustomUserBackend(ModelBackend):
 
     user_models = (Faculty, ResearchScholar, Student)  # order is important
 
-    def authenticate(self, *args, **kwargs):
-        return self.downcast_user_type(super().authenticate(*args, **kwargs))
```
Comment (ujjwal-raizada) [inline backend/users/auth.py:17]: Change the doc string according to this change,

File: backend/users/auth.py [hunk 2]
```diff
@@ -24,6 +21,8 @@ def downcast_user_type(self, user):
         Retrieve the user from the first table that has a row with same email.
         Subclasses should be checked before parents.
         """
+        if user is None:
+            return None
```
Comment (ujjwal-raizada) [inline backend/users/auth.py:25]: in which case 'user' will be None?

**Review by ujjwal-raizada — CHANGES_REQUESTED** [review]

File: backend/users/auth.py (hunk 2 above)
Comment (iamkroot) [reply backend/users/auth.py:25]: AnonymousUser. When no one is logged in.

File: backend/users/auth.py (hunk 1 above)
Comment (iamkroot) [reply backend/users/auth.py:17]: Which docstring?

Comment (ujjwal-raizada) [reply backend/users/auth.py:17]: nvm, I interpreted it wrongly. Doc string is correct.

**Review by ujjwal-raizada — APPROVED** [review]

**Comment by ujjwal-raizada** [comment]
LGTM

**Pull request merged**